compiler/optimizer/DeadTreesElimination.cpp

```diff
@@ -618,6 +653,17 @@ namespace
       };
    }
 
+static bool treeCanNotBeRemoved(TR::Node *node)
```

Comment:
This logic isn't easy to read - I think you could factor these conditions more clearly with some nice comments.